### PR TITLE
Flockdrones stare at birds occasionally

### PR DESCRIPTION
**Pull request by amylizzle**
## About the PR 
Adds a stupid behaviour to occasionally stare at/whistle at/tweet at nearby birds. Should trigger only once every 1-3minutes for any given drone.

## Why's this needed? 
IDK, little easter egg things give the game some character

### CODE REVIEW

File: code/mob/living/critter/ai/flock/flocktasks.dm [hunk 1]
```diff
@@ -905,3 +905,75 @@ butcher
...
+
+/datum/aiTask/succeedable/stare_at_bird/succeeded()
+	return has_started
+
+/datum/aiTask/succeedable/stare_at_bird/on_tick()
+	if(!has_started)
+		var/mob/living/critter/target = holder.target
+		var/mob/living/critter/flock/drone/F = holder.owner
+		if(target && IN_RANGE(holder.owner, target, 1))
+			if(ON_COOLDOWN(holder.owner,"bird_staring",rand(60 SECONDS, 180 SECONDS))) //you can only stare at birds once every now an then. randomise for staggering
```
Comment (TobleroneSwordfish) [inline code/mob/living/critter/ai/flock/flocktasks.dm:951]:
```suggestion
			if(!ON_COOLDOWN(holder.owner,"bird_staring",rand(60 SECONDS, 180 SECONDS))) //you can only stare at birds once every now an then. randomise for staggering
```

File: code/mob/living/critter/ai/flock/flocktasks.dm [hunk 2]
```diff
@@ -905,3 +905,75 @@ butcher
...
+	return !IN_RANGE(holder.owner, holder.target, 1)
+
+/datum/aiTask/succeedable/stare_at_bird/succeeded()
+	return has_started
+
+/datum/aiTask/succeedable/stare_at_bird/on_tick()
+	if(!has_started)
+		var/mob/living/critter/target = holder.target
+		var/mob/living/critter/flock/drone/F = holder.owner
+		if(target && IN_RANGE(holder.owner, target, 1))
```
Comment (TobleroneSwordfish) [inline code/mob/living/critter/ai/flock/flocktasks.dm:950]: should probably make these ifs early returns

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
I love everything about this PR

File: code/mob/living/critter/ai/flock/flocktasks.dm (hunk 1 above)
Comment (amylizzle) [reply code/mob/living/critter/ai/flock/flocktasks.dm:951]: oops lol

File: code/mob/living/critter/ai/flock/flocktasks.dm (hunk 2 above)
Comment (amylizzle) [reply code/mob/living/critter/ai/flock/flocktasks.dm:950]: yeah, probably. will fix

**Comment by stonepillars** [comment]
birb

**Pull request merged**